src/components.jl

```
@@ -302,3 +314,49 @@ function get_num_components(components::Components)
     end
     return count
 end
+
+function clear_forecasts!(components::Components)
+    for component in iterate_components_with_forecasts(components)
+        clear_forecasts!(component)
+    end
+end
+
+function get_forecast_initial_times(components::Components)::Vector{Dates.DateTime}
+    initial_times = Set{Dates.DateTime}()
+    for component in iterate_components_with_forecasts(components)
+        get_forecast_initial_times!(initial_times, component)
+    end
+
+    return sort!(Vector{Dates.DateTime}(collect(initial_times)))
+end
+
+function get_forecasts_initial_time(components::Components)
+    initial_times = get_forecast_initial_times(components)
```

it's better to throw an error here or an exception.